### ISSUE TITLE
Bug: Weave predefined scorers fail for `AsyncOpenAI` clients

### ISSUE THREAD

**Issue by TeoZosa** [wandb/weave#3051]
Excited for the addition of new scoring functions, thanks for the hard work. With it being hot off the presses, I did run into this issue.

 `AsyncOpenAI` clients seem to be supported:
https://github.com/wandb/weave/blob/585752e31856f2a67064fe35d91de639f04382b8/weave/scorers/llm_utils.py#L28-L30

But the underlying calls for Weave scorers assume non-async clients:
https://github.com/wandb/weave/blob/585752e31856f2a67064fe35d91de639f04382b8/weave/scorers/llm_utils.py#L85-L96

**Comment by TeoZosa** [comment]
Thanks @tcapelle for jumping on this so fast! 🙏

**Comment by TeoZosa** [comment]
Hey @tcapelle, has this been deprioritized? We've been vendoring a fork of the code in our repos but this issue is blocking feature mainlining and broader release of some of our internal tools.

**Comment by tcapelle** [comment]
Not at all, but we have a significant scorer overhaul incoming, that's why this change has not been integrated yet.
